app/models/festivity_market_list.rb

```diff
@@ -0,0 +1,76 @@
+class FestivityMarketList
+
+  include Concerns::FestivitySqlBuilder
+
+  attr_reader :markets
+
+
+  def initialize(market_details)
+    @markets = market_details
+  end
+
+  def self.search(criteria, order_by)
+    begin
+      where_clause = parse_criteria(criteria)
+    rescue ActiveRecord::RecordNotFound
+      return FestivityMarketList.new([])
+    end
+
+    # where in market ids
+    FestivityMarketList.new(
+      FestivityMarketList::FestivityMarketDetail.
+        includes(:assets).
+        joins(:festivity_categories).
+        where(where_clause).
+        order("#{order_by} ASC").
+        preload(:festivity_categories)
+    )
+  end
+
+
+  private
+
+
+  # Return a list of unique market ids that match the provided dates
+  def self.market_ids_for_datetimes(datetimes, filter_type)
+    FestivityMarketList::FestivityMarketDetail.where(datetime_criteria(datetimes, filter_type)).map {|e| e.market_id}.uniq
+  end
+
+  # Create a condition for start and end date between midnight and 11:59pm
+  # for each date passed in and return the SQL condition
+  def self.datetime_criteria(datetimes_string, filter_type)
+
+    date_queries = datetimes_string.split(',').map do |date_string|
+      start_date = Chronic.parse(URI.decode(date_string)).utc
+      end_date = start_date.advance(advance_by(filter_type))
+      <<-SQL
+        (
+          (start_date <= '#{start_date}' AND end_date >= '#{end_date}')
+          OR
+          (end_date >= '#{start_date}' AND start_date <= '#{end_date}')
+        )
+      SQL
```

Comment:
We'll get some real data in there so we can test it. We'll need to check the timezone on the servers as well.